library/hawkular_alerts_member.py

```diff
@@ -0,0 +1,188 @@
+    required: True
+  id:
+    description:
+      - the group member id
+    default: null
+    required: False
+  data_id_map:
+    description:
+      - a mapping of the token to the real data_id
+    default: null
+    required: True
+  name:
+    description:
+      - the group member name
+    default: null
+    required: False
+  state:
+    description:
+      - the state of the user
+      - On present, it will create the group trigger
+      if it does not exist
+    required: True
+    choices: ['present', 'absent']
+  scheme:
+    description:
+      - the hawkular scheme
+    default: 'https'
+    required: False
+    choices: ['https', 'http']
+  cafile:
+    description:
+      - the path to the ca file
+    default: null
+    required: False
+'''
+
+EXAMPLES = '''
+# create group member
+  hawkular_alerts_member:
+    hawkular_api_hostname: 'hawkular-endpoint.example.com'
+    port: 443
+    token: '******'
+    tenant: '_system'
+    state: 'present'
+    group_id: 'example-group-trigger'
+    id: 'member1'
+    name: 'Member One'
+    data_id_map:
+      my-metric-id: my-metric-id-member1
+'''
+
+import urllib2
+import hawkular.alerts
+
+# TODO: remove once the ssl cert verification is checked
+import os
+
+
+class HawkularAlertsGroupMember(object):
+    """ Hawkular Alerts object to create group members in Hawkular
+    """
+    def __init__(self, module, tenant, hostname, port, scheme, token, context):
+        self.module  = module
+        self.client  = hawkular.alerts.HawkularAlertsClient(tenant, host=hostname, port=port, scheme=scheme, token=token, context=context)
+        self.changed = False
+
+    def group_trigger_exist(self, group_id):
+        """
+            Returns:
+                True if a group trigger with the passed id exist, False otherwise
+        """
+        try:
+            self.client.get_trigger(group_id)
+            return True
+        except urllib2.HTTPError as err:
+            if err.code == 404:
+                return False
+            else:
+                self.module.fail_json(msg="Failed to get group trigger. Error: {error}".format(error=err))
+
+    def create_group_member(self, group_id, member_id, data_id_map, member_name=None):
+        """ Creates a group member in Hawkular Alerting component
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail_json(msg="Failed to create group member, group {group_id} does not exist ".format(group_id=group_id))
+        try:
+            #  create group member object
+            member = hawkular.alerts.GroupMemberInfo()
+            member.group_id = group_id
+            member.member_id = member_id
+            member.member_name = member_name
+            member.data_id_map = data_id_map
+            self.client.create_group_member(member)
```

Comment:
I'm introducing the idempotency in https://github.com/dkorn/hawkular-alerts-module/pull/5,
since we depend on https://github.com/hawkular/hawkular-client-python/pull/27